apps/examples/websocket/websocket_main.c

```diff
@@ -782,19 +777,23 @@ int websocket_main(int argc, char *argv[])
 			goto error_with_function_failure;
 		}
 		pthread_setname_np(tid, "websocket client");
-		pthread_detach(tid);
+		pthread_join(tid, &exit_value);
 	} else if (memcmp(argv[1], "server", strlen("server")) == 0 && argc == 3) {
 		if ((status = pthread_create(&tid, &attr, (pthread_startroutine_t)websocket_server, (void *)input)) != 0) {
 			printf("fail to create thread\n");
 			goto error_with_function_failure;
 		}
 		pthread_setname_np(tid, "websocket server");
-		pthread_detach(tid);
+		pthread_join(tid, &exit_value);
 	} else {
 		printf("\nwrong input parameter !!!\n %s\n", WEBSOCKET_USAGE);
 		goto error_with_function_failure;
 	}
 
+	if (input) {
```

Comment:
Do we need to check whether input has value or not?
I think free(input); is enough.
Previous codes and current codes either set input to NULL. This conditional always true I think. Let's reduce a code size.

Reply:
@sunghan-chang , you are right that we can reduce the code size by not having the conditional checks in this context